src/Cli/ConfigGenerator.cs

```diff
@@ -510,6 +510,72 @@ public static bool TryAddNewEntity(AddOptions options, RuntimeConfig initialRunt
             return true;
         }
 
+        /// <summary>
+        /// Tries to update the runtime settings based on the provided runtime options.
+        /// </summary>
+        /// <returns>True if the update was successful, false otherwise.</returns>
+        public static bool TryConfigureSettings(ConfigureOptions options, FileSystemRuntimeConfigLoader loader, IFileSystem fileSystem)
+        {
+            if (!TryGetConfigFileBasedOnCliPrecedence(loader, options.Config, out string runtimeConfigFile))
+            {
+                return false;
+            }
+
+            if (!loader.TryLoadConfig(runtimeConfigFile, out RuntimeConfig? runtimeConfig))
+            {
+                _logger.LogError("Failed to read the config file: {runtimeConfigFile}.", runtimeConfigFile);
+                return false;
+            }
+
+            if (options.DepthLimit is not null && !TryUpdateDepthLimit(options, ref runtimeConfig))
+            {
+                return false;
+            }
+
+            return WriteRuntimeConfigToFile(runtimeConfigFile, runtimeConfig, fileSystem);
+        }
+
+        /// <summary>
+        /// Attempts to update the depth limit in the GraphQL runtime settings based on the provided value.
+        /// Validates that any user-provided depth limit is an integer within the valid range of [1 to Int32.MaxValue] or -1.
+        /// A depth limit of -1 is considered a special case that disables the GraphQL depth limit.
+        /// </summary>
+        /// <param name="options">Options including the new depth limit.</param>
+        /// <param name="runtimeConfig">Current config, updated if method succeeds.</param>
+        /// <returns>True if the update was successful, false otherwise.</returns>
+        private static bool TryUpdateDepthLimit(
```

Comment:
What happens if someone calls this function and the options.DepthLimit is null?

Reply:
we already had a check before calling this method, that only calls this when depthLimit is not null. But in future if some other method wants to call it, for them to be aware, i will update the method summary.